### PR TITLE
[OpenJPEG] Update to 2.5.0

### PR DESCRIPTION
**Pull request by Neumann-A**
~~Blocked on: https://github.com/uclouvain/openjpeg/issues/1422~~ fixed it myself and opened a PR upstream

### CODE REVIEW

File: ports/openjpeg/portfile.cmake [hunk 1]
```diff
@@ -31,6 +24,9 @@ vcpkg_cmake_configure(
         -DBUILD_PKGCONFIG_FILES=ON
         ${FEATURE_OPTIONS}
         -DBUILD_STATIC_LIBS=${BUILD_STATIC_LIBS}
+        # -DBUILD_JPIP_SERVER=ON # Requires curl
+        # -DBUILD_VIEWER=ON 
+        # -DBUILD_CODEC=ON # requires ZLIB; Tiff; LCMS2; wxWidgets?
```
Comment (Neumann-A) [inline ports/openjpeg/portfile.cmake:29]: Just documenting if somebody wants to add a feature `tools`

**Comment by Neumann-A** [comment]
> You have modified or added at least one vcpkg.json where you should check the license field.

Somebody mind telling me what type of BSD that is ? (https://github.com/uclouvain/openjpeg/blob/master/LICENSE)

**Comment by rouault** [comment]
> Somebody mind telling me what type of BSD that is ?

it is written in the license :-) "2-clauses  BSD License" ==> https://spdx.org/licenses/BSD-2-Clause.html

File: ports/openjpeg/portfile.cmake [hunk 2]
```diff
@@ -53,17 +51,8 @@ vcpkg_fixup_pkgconfig()
...
-    list(APPEND TOOL_NAMES opj_compress opj_decompress opj_dump opj_mj2_compress opj_mj2_decompress opj_mj2_extract opj_mj2_wrap)
-endif()
-if("jpip" IN_LIST FEATURES)
-    list(APPEND TOOL_NAMES opj_compress opj_decompress opj_dump opj_dec_server opj_jpip_addxml opj_jpip_test opj_jpip_transcode)
-endif()
-if("jp3d" IN_LIST FEATURES)
-    list(APPEND TOOL_NAMES opj_jp3d_compress opj_jp3d_decompress)
+if("jpip" IN_LIST FEATURES) # Now behind BUILD_VIEWER / BUILD_CODEC / BUILD_JPIP_SERVER
+    #list(APPEND TOOL_NAMES opj_compress opj_decompress opj_dump opj_dec_server opj_jpip_addxml opj_jpip_test opj_jpip_transcode)
 endif()
```
Comment (Adela0814) [inline ports/openjpeg/portfile.cmake:56]: Is this necessary?

File: ports/openjpeg/vcpkg.json [hunk 3]
```diff
@@ -15,16 +16,16 @@
   ],
   "features": {
     "jp3d": {
-      "description": "Build optional component jp3d"
+      "description": "(deprecated)"
```
Comment (Adela0814) [inline ports/openjpeg/vcpkg.json:19]: You can delete these deprecated features.

**Review by Adela0814 — CHANGES_REQUESTED** [review]

File: ports/openjpeg/vcpkg.json (hunk 3 above)
Comment (Neumann-A) [reply ports/openjpeg/vcpkg.json:19]: I cannot delete already existing features since it would break `vcpkg upgrade`.

**Review by Adela0814 — APPROVED** [review]

**Comment by BillyONeal** [comment]
Upstream merged the thing https://github.com/uclouvain/openjpeg/pull/1423/files

**Comment by BillyONeal** [comment]
Thanks for the update!

**Pull request merged**